packages/store/test/test-pattern-limits.js

```diff
@@ -0,0 +1,243 @@
+    noPatterns: [
+      [
+        M.symbol(harden({ symbolNameLengthLimit: 10 })),
+        'Symbol name "moderate length string" must not be bigger than 10',
+      ],
+    ],
+  },
+  {
+    specimen: Symbol.for('x'.repeat(defaultLimits.symbolNameLengthLimit + 1)),
+    yesPatterns: [M.symbol(harden({ symbolNameLengthLimit: Infinity }))],
+    noPatterns: [
+      [
+        M.symbol(),
+        'Symbol name "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx" must not be bigger than 40',
+      ],
+    ],
+  },
+  // numPropertiesLimit, propertyNameLengthLimit
+  {
+    specimen: {
+      z: 1000000n,
+      x0123456789: 379n,
+      a: 10000000n,
+    },
+    yesPatterns: [
+      M.record(),
+      M.record(harden({ numPropertiesLimit: 10, propertyNameLengthLimit: 20 })),
+      M.recordOf(M.string(), M.bigint()),
+      M.recordOf(
+        M.string(),
+        M.bigint(),
+        harden({ numPropertiesLimit: 10, propertyNameLengthLimit: 20 }),
+      ),
+    ],
+    noPatterns: [
+      [
+        M.record(harden({ numPropertiesLimit: 2 })),
+        'Must not have more than 2 properties: {"z":"[1000000n]","x0123456789":"[379n]","a":"[10000000n]"}',
+      ],
+      [
+        M.record(harden({ propertyNameLengthLimit: 5 })),
+        'x0123456789: Property name must not be longer than 5',
+      ],
+    ],
+  },
+  // arrayLengthLimit
+  {
+    specimen: [...'moderate length string'],
+    yesPatterns: [
+      M.array(),
+      M.arrayOf(M.string()),
+      M.array(harden({ arrayLengthLimit: 40 })),
+      M.arrayOf(M.string(), harden({ arrayLengthLimit: 40 })),
+    ],
+    noPatterns: [
+      [
+        M.array(harden({ arrayLengthLimit: 10 })),
+        'Array length 22 must be <= limit 10',
+      ],
+      [M.arrayOf(M.number()), '[0]: string "m" - Must be a number'],
+      [
+        M.arrayOf(M.number(), harden({ arrayLengthLimit: 10 })),
+        'Array length 22 must be <= limit 10',
+      ],
+      [
+        M.arrayOf(M.string(), harden({ arrayLengthLimit: 10 })),
+        'Array length 22 must be <= limit 10',
+      ],
+    ],
+  },
+  {
+    specimen: Array(defaultLimits.arrayLengthLimit + 1).fill(1),
+    yesPatterns: [M.array(harden({ arrayLengthLimit: Infinity }))],
+    noPatterns: [[M.array(), 'Array length 10001 must be <= limit 10000']],
+  },
+  // numSetElementsLimit
+  {
+    specimen: makeCopySet([0, 1, 2, 3, 4, 5]),
+    yesPatterns: [M.set(), M.setOf(M.number())],
+    noPatterns: [
+      [
+        M.set(harden({ numSetElementsLimit: 3 })),
+        'Set must not have more than 3 elements: 6',
+      ],
+    ],
+  },
+  // numUniqueBagElementsLimit
+  {
+    specimen: makeCopyBag([
+      [0, 37n],
+      [1, 3n],
+      [2, 100000n],
+      [3, 1n],
+    ]),
+    yesPatterns: [M.bag(), M.bagOf(M.number())],
+    noPatterns: [
+      [
+        M.bag(harden({ numUniqueBagElementsLimit: 3 })),
+        'Bag must not have more than 3 unique elements: "[copyBag]"',
```

Comment:
fix non-uniformity of error message between 206 and 222. Make all follow the winner.

Reply:
Postponed past this PR https://github.com/Agoric/agoric-sdk/issues/6149